crates/cli-support/src/js/js2rust.rs

```diff
@@ -2,6 +2,22 @@ use crate::descriptor::{Descriptor, Function};
 use crate::js::Context;
 use failure::{bail, Error};
 
+pub struct JsArgument {
+    pub optional: bool,
+    pub name: String,
+    pub type_: String,
+}
+
+impl JsArgument {
+    fn required(name: String, type_: String) -> Self {
+        Self { optional: false, name, type_ }
+    }
+
+    fn optional(name: String, type_: String) -> Self {
+        Self { optional: true, name, type_ }
+    }
+}
```

Comment:
Should the names of these methods be changed to new_(required|optional)?

Reply:
Are there any guidelines / recommendations on this topic? I couldn't find anything in https://github.com/rust-dev-tools/fmt-rfcs/.

I personally prefer the shorter version, it's clear from the signature that it's a constructor.